src/mca/odls/base/odls_base_default_fns.c

```diff
@@ -904,13 +904,16 @@ void prrte_odls_base_spawn_proc(int fd, short sd, void *cbdata)
     cd->env = prrte_argv_copy(prrte_launch_environ);
     if (NULL != app->env) {
         for (i=0; NULL != app->env[i]; i++) {
-            /* find the '=' sign */
-            ptr = strchr(app->env[i], '=');
+            /* find the '=' sign.
+             * strdup the env string, since it
+             * is shared among apps.
+             */
+            char *ptr_ = strdup(app->env[i]);
```

Comment:
ptr is already a declared variable - let's not redeclare it here

Reply:
Oh, I see - you declared another variable! Let's call it something a little more obvious as it already confused me.

Reply:
maybe `key` since that's what it represents to the setenv?

Reply:
I don't care - just something more obviously different than `ptr`

Reply:
I changed it to `tmp`. Is that ok?

Reply:
sure - thx!